### ISSUE TITLE
Questions abou the weight sharing method in GNN/Pyg

### ISSUE THREAD

**Issue by HelloWorldLTY** [pyg-team/pytorch_geometric#6285]
### 🚀 The feature, motivation and pitch

Hi, I intend to implement the weight sharing method in pyg. I notice that there is a post which is related to this work:
https://github.com/pyg-team/pytorch_geometric/issues/1503
However, for other module like GATconv and Transformconv, I did not find conv.weight or conv.bias. Could you please help me figure out why? Thanks.

**Comment by HelloWorldLTY** [comment]
Hi, I just notice that we can use state_dict to find the weight of each component. So now my question is, if I intend to set shared parameter, is it enough for me to set it once, for example (m1.weight = m0.weight + m1.weight)? Or I need to include this update  for each forward step? Thanks.

**Comment by rusty1s** [comment]
Do you mean weight sharing across layers? For this, you can simply re-use the GNN layer across iterations, e.g.:
```python
for _ in range(3):
    x = self.conv(x, edge_index)

**Comment by HelloWorldLTY** [comment]
Hi, thanks for your explaniation. I actually intend to modify each layers' weights, which are based on other layers weight. For example, if I have three layer network, the first layer's weight will be l1.weight+l2.weight+l3.weight. Is it possible to implement it? Thanks.

**Comment by rusty1s** [comment]
Yes, then you can obtain the `state_dict`, modify the weights to your desire, and update them using `load_state_dict`. But this will not share the weights, it will just initialize them as you want.

**Comment by HelloWorldLTY** [comment]
Thanks. Therefore, if I design common weight in a model, using nn.Parameters, then assign this common weight to different state, will it be a possible implement? Thanks.

**Comment by rusty1s** [comment]
Yes, this should be possible. You can look into what needs to be replaced exactly by looking at the source code of `GATConv`.

**Comment by HelloWorldLTY** [comment]
Thanks a lot.
